modules/uac/replace.c

```diff
@@ -270,8 +270,19 @@ int replace_uri( struct sip_msg *msg, str *display, str *uri,
 		if ( body->display.len) {
 			LM_DBG("removing display [%.*s]\n",
 				body->display.len,body->display.s);
+
+			start_offset = body->display.s-msg->buf;
+
+			//If replacing leave trailing spaces
+			if (display->len) {
+				display_len = body->display.len;
+			} else {
+			//Removing, strip all characters up to the URI
+				display_len = ((body->uri.s-msg->buf) - start_offset) - 1;
```

Comment:
I guess the `-1` is for leaving a single space between the URI and the display, right ? If so, I'm not sure that the case of an enclosed URI is properly handled. If I'm not wrong, the `<` and `>` are not part of the `uri` field.
Also, the patch does not cover the situation of multiple spaces **before** the display, right?

Reply:
You are correct that `<>` are not part of the uri field.

The `-1` actually stops us from trimming the `<` before the `uri`. I believe a header with a display name, but a non-enclosed `uri` is against spec (https://www.ietf.org/rfc/rfc2822.txt), so it shouldn't be a problem?

We remove everything from the start of the display name up to, but not including the `<`. What is left are the spaces from before the display name and the `<>` enclosed `uri`.

The patch **does not** cover multiple spaces **before** the display, but it could be modified to include those as well.

Example behaviour (verified against a test):

Original:
`From: "Name" <sip:user@domain>;tag=as2ed502bc`
Stripped:
`From: <sip:user@domain>;tag=as2ed502bc`